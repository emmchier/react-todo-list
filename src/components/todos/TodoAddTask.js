import React from 'react';

import { CustomBtn } from '../../ui/customs/CustomBtn';
import { CustomFormField } from '../../ui/customs/CustomFormField';

export const TodoAddTask = () => {

    const handleAddTask = (e) => {
        e.preventDefault();
        console.log('Nueva tarea agregada');
    }

    return (
        <form onSubmit={ handleAddTask } className="home__add-form">
            <div className="home__form-content alignX">
                <CustomFormField 
                    formInputType={ 'text' }
                    formInputName={ 'desc' }
                    formInputPlaceholder={ 'Nueva tarea...' }
                /> 
                <CustomBtn
                    btnTitle={ 'Agregar' }
                    classes={ 'btn-add btnNormal' }
                    btnType={ 'submit' }
                />
            </div>
        </form>
    )
}