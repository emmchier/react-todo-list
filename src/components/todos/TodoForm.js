import React from 'react';
import { CustomBtn } from 'ui/customs/CustomBtn';
import { CustomFormField } from 'ui/customs/CustomFormField';

export const TodoForm = ({
    inputType,
    inputValue,
    inputName,
    inputOnChange,
    inputPlaceholder,
    btnTitle,
    btnClasses,
    btnType,
    onSubmit
}) => {
    return (
        <form onSubmit={onSubmit}>
            <div className="alignX">
                <CustomFormField
                    formInputType={inputType}
                    formInputName={inputName}
                    formInputValue={inputValue}
                    formInputPlaceholder={inputPlaceholder}
                    onChange={inputOnChange}
                />
                <CustomBtn
                    btnTitle={btnTitle}
                    classes={btnClasses}
                    btnType={btnType}
                />
            </div>
        </form>
    )
}
