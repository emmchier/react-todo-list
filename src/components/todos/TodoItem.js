import React from 'react';

import { CustomBtn } from '../../ui/customs/CustomBtn';

export const TodoItem = ({task, index, dispatch}) => {

    const handleDeleteTask = () => {

        dispatch({
            type: 'delete',
            payload: task.id
        });
    }

    const handleCheckedTask = () => {

        dispatch({
            type: 'checked',
            payload: task.id
        });
    }

    return (
        <li
            className="home__todo-item list-group-item"
        >
            <div className="home__todo-item-content alignX pushAside">
                <p 
                    className={ `home__todo-desc ${ task.done && 'checkedTask' }` }
                    onClick={ handleCheckedTask }>
                    { index + 1 }. { task.desc }    
                </p>    
                <CustomBtn
                    btnTitle={ 'Borrar' }
                    classes={ 'btn-delete btnNormal' }
                    onClick={ handleDeleteTask }
                />
            </div>
        </li>
    )
}
