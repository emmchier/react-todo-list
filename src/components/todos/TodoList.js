import React from 'react';
import { TodoAddTask } from './TodoAddTask';
import { TodoItem } from './TodoItem';

export const TodoList = ({
    todos, 
    dispatch, 
    hideCompleted
}) => {

    const showCompleted = hideCompleted ? todos.filter(task => !task.done) : todos;

    return (
        <>
            <div className="home__todoList">
                <div className="todoList__header">
                    <div className="row pushAside">
                        <div className="col-sm col-md-6">
                            <h3>Mis tareas ({todos.length})</h3>
                        </div>
                        <div className="col-sm col-md-6">
                            <TodoAddTask
                                todoList={todos}
                                dispatch={dispatch}
                            />
                        </div>
                    </div>
                </div>
                <ul className="list-group-flush">
                    {
                        showCompleted.map((task, i) => (
                            <TodoItem
                                key={task.id}
                                index={i}
                                task={task}
                                dispatch={dispatch}
                            />
                        ))
                    }
                </ul>
            </div>
           
        </>
    )
}
