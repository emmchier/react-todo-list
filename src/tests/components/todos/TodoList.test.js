import { TodoList } from 'components/todos/TodoList';
import { demoTodos } from 'tests/fixtures/demoTodos';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

describe('Pruebas en <TodoList />', () => {

    const dispatch = jest.fn();

    const wrapper = shallow(
        <TodoList
            todos={demoTodos}
            dispatch={dispatch}
            hideCompleted={false}
        />
    )

    test('debe mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe contener el total de sus items actuales', () => {
        expect(wrapper.find('TodoItem').length).toBe(demoTodos.length);
    })
});