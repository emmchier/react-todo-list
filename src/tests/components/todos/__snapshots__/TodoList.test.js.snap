// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> debe mostrarse correctamente  1`] = `ShallowWrapper {}`;
